src/js/components/AppListFilterComponent.jsx

```diff
@@ -0,0 +1,88 @@
+var classNames = require("classnames");
+var React = require("react/addons");
+
+var AppListFilterComponent = React.createClass({
+  displayName: "AppListFilterComponent",
+
+  propTypes: {
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      filterText: "",
+      activated: false,
+      focused: false
+    };
+  },
+
+  updateFilterText: function (event) {
+    var filterText = event.target.value;
+    this.setState({
+      filterText: filterText,
+      activated: filterText !== "" || this.state.focused
+    });
+    this.props.onChange(filterText);
+  },
+
+  clearFilterText: function () {
+    this.setState({
+      filterText: "",
+      activated: false
+    });
+    this.props.onChange("");
+  },
+
+  getFilterBox: function () {
+    var state = this.state;
+
+    var filterBoxClassSet = {
+      "input-group": true,
+      "filter-box": true,
+      "filter-box-activated": !!state.activated,
+      "pull-left": true
```

Comment:
That's what I thought as well, however with only the button floated right, they no longer align vertically.  This is because bootstrap uses `display: table` for input groups.  Not using input groups would be a larger change (since thats how I implemented the icons inside the input).

I'm not great at css, so other suggestions are definitely welcome.